src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -471,9 +463,39 @@ bool Terminal::SendMouseEvent(const COORD viewportPos, const unsigned int uiButt
     return _terminalInput->HandleMouse(viewportPos, uiButton, GET_KEYSTATE_WPARAM(states.Value()), wheelDelta);
 }
 
-bool Terminal::SendCharEvent(const wchar_t ch)
+// Method Description:
+// - Send this particular character to the terminal.
+// - This method is the counterpart to SendKeyEvent and behaves almost identical.
+//   The difference is the focus on sending characters to the terminal,
+//   whereas SendKeyEvent handles the sending of keys like the arrow keys.
+// Arguments:
+// - ch: The UTF-16 code unit to be sent.
+// - scanCode: The scan code of the last pressed key. Can be left 0.
+// - states: The Microsoft::Terminal::Core::ControlKeyStates representing the modifier key states.
+// Return Value:
+// - true if we translated the character event, and it should not be processed any further.
+// - false otherwise.
+bool Terminal::SendCharEvent(const wchar_t ch, const WORD scanCode, const ControlKeyStates states)
 {
-    return _terminalInput->HandleChar(ch);
+    // DON'T manually handle Alt+Space - the system will use this to bring up
+    // the system menu for restore, min/maximize, size, move, close.
+    if (ch == L' ' && states.IsAltPressed() && !states.IsCtrlPressed())
+    {
+        return false;
+    }
+
+    auto vkey = _TakeVirtualKeyFromLastKeyEvent(scanCode);
+    if (vkey == 0 && scanCode != 0)
+    {
+        vkey = _ScanCodeFromVirtualKey(scanCode);

```

Comment:
this name worries me. Below, we have `VirtualKeyFromCharacter`, but here we're _assigning_ `vkey` but the function name implies it returns a ScanCode

Reply:
You‘re right: This should be `_VirtualKeyFromScanCode`. 👍